tests/tt_metal/tt_metal/test_compile_sets_kernel_binaries.cpp

```diff
@@ -200,15 +200,19 @@ int main(int argc, char **argv) {
                             JitBuildProcessorType::DATA_MOVEMENT,
                             0,
                             get_latest_kernel_binary_path(mask, riscv0_kernel));
-                        ll_api::memory brisc_binary = llrt::get_risc_binary(brisc_hex_path, 0, llrt::PackSpans::PACK);
+                        ll_api::memory brisc_binary = llrt::get_risc_binary(brisc_hex_path, 0, ll_api::memory::PackSpans::PACK, ll_api::memory::Relocate::XIP);
                         TT_FATAL(
                             brisc_binary == brisc_binaries.at(mask).at(0),
                             "Expected saved BRISC binary to be the same as binary in persistent cache");
                         std::string ncrisc_hex_path = device->build_kernel_target_path(
                             JitBuildProcessorType::DATA_MOVEMENT,
                             1,
                             get_latest_kernel_binary_path(mask, riscv1_kernel));
-                        ll_api::memory ncrisc_binary = llrt::get_risc_binary(ncrisc_hex_path, 1, llrt::PackSpans::PACK);
+                        ll_api::memory::Relocate relo_type =
+                            (device->arch() == tt::ARCH::GRAYSKULL || device->arch() == tt::ARCH::WORMHOLE_B0) ?
+                            ll_api::memory::Relocate::NONE : ll_api::memory::Relocate::XIP;
```

Comment:
is this to cover case where we can't relocate due to ncrisc iram?

Reply:
Yes.  Should migrate this to the HAL and make an attribute query "has_iram"